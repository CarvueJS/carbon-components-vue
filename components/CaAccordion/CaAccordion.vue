<template>
  <ul data-accordion class="bx--accordion">
    <slot></slot>
  </ul>
</template>

<script>
import { Accordion } from 'carbon-components';

export default {
  name: 'ca-accordion',
  mounted() {
    Accordion.create(this.$el);
  },
};
</script>

<style></style>
