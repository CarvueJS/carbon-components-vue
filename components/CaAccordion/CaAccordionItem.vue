<template>
  <li data-accordion-item class="bx--accordion__item" :class="dynamicClass">
    <button class="bx--accordion__heading" aria-expanded="false" aria-controls="pane1">
      <svg class="bx--accordion__arrow" width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
      </svg>
      <p class="bx--accordion__title">
        {{title}}
      </p>
    </button>
    <div class="bx--accordion__content">
      <slot></slot>
    </div>
  </li>
</template>

<script>
export default {
  name: 'ca-accordion-item',
  props: {
    title: {
      type: String,
      default: 'title',
    },
    active: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    dynamicClass() {
      return {
        'bx--accordion__item--active': this.active,
      };
    },
  },
};
</script>

<style></style>
