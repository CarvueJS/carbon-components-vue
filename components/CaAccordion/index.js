import CaAccordion from './CaAccordion.vue';
import CaAccordionItem from './CaAccordionItem.vue';

function install(Vue) {
  Vue.component(CaAccordion.name, CaAccordion);
  Vue.component(CaAccordionItem.name, CaAccordionItem);
}

export default {
  install,
};
