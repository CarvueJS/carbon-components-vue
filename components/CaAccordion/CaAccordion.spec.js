import { shallow } from 'vue-test-utils';
import CaAccordion from './CaAccordion.vue';
import CaAccordionItem from './CaAccordionItem.vue';

describe('CaAccordion', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaAccordion, { 
      slots: {
        default: CaAccordionItem
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot: pass title props', () => {
    const wrapper = shallow(CaAccordion, { 
      slots: {
        default: {
          render(h) {
            return h(CaAccordionItem, { props: { title: 'titleText' } })
          }
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot()
  });

  test('Mathes snapshot: pass activie props', () => {
    const wrapper = shallow(CaAccordion, { 
      slots: {
        default: {
          render(h) {
            return h(CaAccordionItem, { props: { active: true } })
          }
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot()
  });
})
