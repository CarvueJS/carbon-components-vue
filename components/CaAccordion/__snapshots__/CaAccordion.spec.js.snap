// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaAccordion Mathes snapshot: pass activie props 1`] = `
<ul data-accordion="" class="bx--accordion">
  <li data-accordion-item="" class="bx--accordion__item bx--accordion__item--active">
    <button aria-expanded="false" aria-controls="pane1" class="bx--accordion__heading">
      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd" class="bx--accordion__arrow">
        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
      </svg>
      <p class="bx--accordion__title">
        title
      </p>
    </button>
    <div class="bx--accordion__content"></div>
  </li>
</ul>
`;

exports[`CaAccordion Mathes snapshot: pass title props 1`] = `
<ul data-accordion="" class="bx--accordion">
  <li data-accordion-item="" class="bx--accordion__item">
    <button aria-expanded="false" aria-controls="pane1" class="bx--accordion__heading">
      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd" class="bx--accordion__arrow">
        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
      </svg>
      <p class="bx--accordion__title">
        titleText
      </p>
    </button>
    <div class="bx--accordion__content"></div>
  </li>
</ul>
`;
